src/UI/Components/Diagram/helpers.ts

```diff
@@ -388,8 +395,15 @@ export const bundleInstances = (
   const embeddedInstances = deepCopiedMapToArray.filter(
     (instance) => !topServicesNames.includes(instance.service_entity),
   );
+  return topInstances.map((instance) => {
+    const serviceModel = services.find(
+      (service) => service.name === instance.service_entity,
+    );
 
-  return topInstances.map((instance) =>
-    shapesDataTransform(embeddedInstances, instance),
-  );
+    return shapesDataTransform(
+      embeddedInstances,
+      instance,
+      serviceModel as ServiceModel,
```

Comment:
if you type serviceModel correctly in line 399, you're not supposed to cast it.

Reply:
the reason why it is casted as I need to find the serviceModel from the array to pass down into the function, there it becomes `serviceModel | undefined`, I can add if statement and then filter out possible undefined values from array, but theoretically we should have all serviceModels in the first place, that why I chose this solution